src/App/LoadFigure.purs

```diff
@@ -55,22 +55,39 @@ loadSpec filename = do
       Left err -> error ("Json fetching failed with " <> printError err)
       Right response -> pure $ response.body
 
-loadFigureFromJson :: Json -> Effect Unit
-loadFigureFromJson json = runAffs_ (uncurry drawFig)
-   [ case decodeJson json :: Either JsonDecodeError JsonSpec of
-        Left err -> error ("JSON decoding failed with " <> show err)
-        Right spec -> ("fig" × _) <$> runWebT (FileCxt { fluidSrcPaths }) (loadFig figSpec)
-           where
-           figSpec@{ fluidSrcPaths } = figSpecFromJson spec
-   ]
+loadFigure :: String -> String -> Effect Unit
+loadFigure specFile srcFile = launchAff_ do
+   jsonSpec <- loadSpec specFile
+   liftEffect $ loadFigureSpec jsonSpec srcFile
 
-loadFigure :: String -> Effect Unit
-loadFigure filename = launchAff_ do
-   jsonSpec <- loadSpec filename
-   liftEffect $ loadFigureFromJson jsonSpec
+loadFigureSrc :: String -> String -> Effect Unit
+loadFigureSrc specFile fluidSrc = launchAff_ do
+   jsonSpec <- loadSpec specFile
+   liftEffect $ loadFigureSpecSrc jsonSpec fluidSrc
 
-drawCode :: String -> String -> Effect Unit
-drawCode folder file = runAffs_ drawFile
-   [ runWebT (FileCxt { fluidSrcPaths: [ Folder folder ] }) $ loadFile_ [ Folder folder ] (File file)
+loadFigureSpec :: Json -> String -> Effect Unit
+loadFigureSpec jsonSpec srcFile = launchAff_ do
+   fluidSrc <- loadFileFromPath (File srcFile)
+   liftEffect $ loadFigureSpecSrc jsonSpec (definitely' fluidSrc)
+
+loadFigureSpecSrc :: Json -> String -> Effect Unit
+loadFigureSpecSrc jsonSpec fluidSrc = runAffs_ (uncurry drawFig)
+   [ case decodeJson jsonSpec :: Either JsonDecodeError JsonSpec of
+        Left err -> error ("JSON decoding failed with " <> show err)
+        Right spec -> do
+           let figSpec@{ fluidSrcPaths } = figSpecFromJson spec
+           ("fig" × _) <$> runWebT (FileCxt { fluidSrcPaths }) (loadFig figSpec fluidSrc)
    ]
 
+drawCode :: String -> Effect Unit
+drawCode file = launchAff_ do
+   fluidSrc <- loadFileFromPath (File file)
+   liftEffect $ drawFile (File (toFileName file) × definitely' fluidSrc)
+   where
+   toFileName :: String -> String
+   toFileName filepath =
```

Comment:
No need to pass `filepath` as an argument -- because `toFileName` (which you could just call `fileName`) is local to `drawCode`, you can just refer directly to `file`, which is in scope.